File: Components/DevExpressVCL/Demos/ExpressQuantumTreeList/Delphi/ColumnsMultiEditorsDemo/ColumnsMultiEditorsDemoData.pas
unit ColumnsMultiEditorsDemoData;

interface

uses
  SysUtils, Classes, DB, Forms, cxClasses, cxStyles, cxTL, dxmdaset;

type
  TColumnsMultiEditorsDemoDataDM = class(TDataModule)
    dsPersons: TDataSource;
    StyleRepository: TcxStyleRepository;
    cxStyle1: TcxStyle;
    cxStyle2: TcxStyle;
    cxStyle3: TcxStyle;
    cxStyle4: TcxStyle;
    cxStyle5: TcxStyle;
    cxStyle6: TcxStyle;
    cxStyle7: TcxStyle;
    cxStyle8: TcxStyle;
    cxStyle9: TcxStyle;
    cxStyle10: TcxStyle;
    cxStyle11: TcxStyle;
    cxStyle12: TcxStyle;
    cxStyle13: TcxStyle;
    stlGroupNode: TcxStyle;
    stlFixedBand: TcxStyle;
    TreeListStyleSheetDevExpress: TcxTreeListStyleSheet;
    mdPersons: TdxMemData;
    mdPersonsID: TAutoIncField;
    mdPersonsName: TStringField;
    procedure DataModuleCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  ColumnsMultiEditorsDemoDataDM: TColumnsMultiEditorsDemoDataDM;

implementation

{$R *.dfm}

procedure TColumnsMultiEditorsDemoDataDM.DataModuleCreate(Sender: TObject);
const
  DataPath = '..\..\Data\';
begin
  mdPersons.LoadFromBinaryFile(DataPath + 'Persons.dat');
  mdPersons.Open;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumTreeList/CBuilder/ColumnsMultiEditorsDemo/ColumnsMultiEditorsDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ColumnsMultiEditorsDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxStyles"
#pragma link "cxTL"
#pragma link "dxmdaset"
#pragma resource "*.dfm"
TColumnsMultiEditorsDemoDataDM *ColumnsMultiEditorsDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TColumnsMultiEditorsDemoDataDM::TColumnsMultiEditorsDemoDataDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TColumnsMultiEditorsDemoDataDM::DataModuleCreate(TObject *Sender)

{
  mdPersons->LoadFromBinaryFile("..\\..\\Data\\Persons.dat");
  mdPersons->Open();
}
//---------------------------------------------------------------------------

